### PR TITLE
ajax

### PR DESCRIPTION
**Pull request by DzmitryKosmach**

### CODE REVIEW

File: task_ajax/js/app.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"use strict";
+
+var API_KEY = "61d6600170ff0cf825a3ee62d4bd7fa6";
+var format = "json";
+new LastFmRequest("chart.gettopartists", null , null, API_KEY, format);
+new LastFmRequest("artist.search", "cher" , null ,API_KEY, format);
```
Comment (NikitaSedyx) [inline task_ajax/js/app.js]: codestyle

File: task_ajax/js/classes/lastFmRequest.class.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+function LastFmRequest(method, artist, album, api_key, format) {
+  var url = "http://ws.audioscrobbler.com/2.0";
+  var method = method ? "/?method=" + method : "";
+  var artist = artist ? "&artist=" + artist : "";
+  var album = album ? "&album=" + album : "";
+  var apiKey = api_key ? "&api_key=" + api_key : "";
+  var format = format ? "&format=" + format : "";
+  this.load(new QueryParams("GET", url + method + artist + album + apiKey + format));
```
Comment (NikitaSedyx) [inline task_ajax/js/classes/lastFmRequest.class.js]: why constructor loads data?

Comment (NikitaSedyx) [inline task_ajax/js/classes/lastFmRequest.class.js]: what about encoding?

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]
you will have problems with this approach when you start drawing the data

**Comment by DzmitryKosmach** [comment]
Last.fm fixed:

https://github.com/DzmitryKosmach/DzmitryKosmach.github.io/pull/6/commits/db47326c6ea9f327c63edbe06c0f7b70d1d62a7c


2017-05-02 19:51 GMT+03:00 Mikita Siadykh <notifications@github.com>:

File: task_ajax/js/classes/albumGetInfo.class.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+function AlbumGetInfo(artist, album, api_key) {
```
Comment (NikitaSedyx) [inline task_ajax/js/classes/albumGetInfo.class.js]: why artist and album are instance fields? they should be arguments for load function, don't you think so?
you no need to create new instance when you want to load data with another parameters

File: task_ajax/js/app.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+"use strict";
+
+var apiKey = "61d6600170ff0cf825a3ee62d4bd7fa6";
```
Comment (NikitaSedyx) [inline task_ajax/js/app.js]: imo it's common parameter

File: task_ajax/js/classes/lastFmRequest.class.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+function LastFmRequest(reguestMethod, artist, album, api_key) {
+  var url = "http://ws.audioscrobbler.com/2.0";
+  var reguestMethod = reguestMethod ? "/?method=" + reguestMethod : "";
+  var artist = artist ? "&artist=" + artist : "";
+  var album = album ? "&album=" + album : "";
+  var apiKey = api_key ? "&api_key=" + api_key : "";
+  var format = "&format=json";
+  this.queryParams = url + reguestMethod + artist + album + apiKey + format;
```
Comment (NikitaSedyx) [inline task_ajax/js/classes/lastFmRequest.class.js]: queryParams requires encodind

File: task_ajax/js/app.js (hunk 4 above)
Comment (NikitaSedyx) [reply task_ajax/js/app.js]: static field of LastFmRequest

**Review by NikitaSedyx — CHANGES_REQUESTED** [review]

**Comment by DzmitryKosmach** [comment]
Please explain the remark:

queryParams requires encodind

In task_ajax/js/classes/lastFmRequest.class.js
<https://github.com/DzmitryKosmach/DzmitryKosmach.github.io/pull/6#discussion_r115035751>
:

> @@ -0,0 +1,24 @@
+function LastFmRequest(reguestMethod, artist, album, api_key) {
+  var url = "http://ws.audioscrobbler.com/2.0";
+  var reguestMethod = reguestMethod ? "/?method=" + reguestMethod : "";
+  var artist = artist ? "&artist=" + artist : "";
+  var album = album ? "&album=" + album : "";
+  var apiKey = api_key ? "&api_key=" + api_key : "";
+  var format = "&format=json";
+  this.queryParams = url + reguestMethod + artist + album + apiKey + format;



2017-05-05 19:21 GMT+03:00 Mikita Siadykh <notifications@github.com>:

**Comment by DzmitryKosmach** [comment]
Last.fm fixed:

https://github.com/DzmitryKosmach/DzmitryKosmach.github.io/pull/6/commits/749cab31309c2c1b88b8d117fe87faaacd151d06


2017-05-05 23:31 GMT+03:00 Dzmitry Kosmach <dzmitry.kosmach@gmail.com>:

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**